packages/upload-media/README.md

```diff
@@ -0,0 +1,147 @@
+# `@wordpress/upload-media`
+
+Core media upload logic implemented with a custom `@wordpress/data` store.
+
+## API Reference
+
+### Actions
+
+The following set of dispatching action creators are available on the object returned by `wp.data.dispatch( 'core/upload-media' )`:
+
+<!-- START TOKEN(Autogenerated actions|src/store/actions.ts) -->
+
+#### addItems
+
+Adds a new item to the upload queue.
+
+_Parameters_
+
+-   _$0_ `AddItemsArgs`:
+-   _$0.files_ `AddItemsArgs[ 'files' ]`: Files
+-   _$0.onChange_ `[AddItemsArgs[ 'onChange' ]]`: Function called each time a file or a temporary representation of the file is available.
+-   _$0.onSuccess_ `[AddItemsArgs[ 'onSuccess' ]]`: Function called after the file is uploaded.
+-   _$0.onBatchSuccess_ `[AddItemsArgs[ 'onBatchSuccess' ]]`: Function called after a batch of files is uploaded.
+-   _$0.onError_ `[AddItemsArgs[ 'onError' ]]`: Function called when an error happens.
+-   _$0.additionalData_ `[AddItemsArgs[ 'additionalData' ]]`: Additional data to include in the request.
+
+#### cancelItem
+
+Cancels an item in the queue based on an error.
+
+_Parameters_
+
+-   _id_ `QueueItemId`: Item ID.
+-   _error_ `Error`: Error instance.
+-   _silent_ Whether to cancel the item silently, without invoking its `onError` callback.
+
+<!-- END TOKEN(Autogenerated actions|src/store/actions.ts) -->
```

Comment:
So by reading the selectors and actions available. What's unclear to me is the "workflow" that the items go through.

I see there's an approval state, a pending state, is uploading...

I think what could be helpful is some kind of schema or documentation about the workflow. WDYT. 

As a user of the package, I only see two actions `addItem` and `cancelItem`, so it's a bit unclear as well how to "approve pending items" or is that internal?

Reply:
Please see https://github.com/swissspidy/media-experiments/blob/main/docs/technical-overview.md#the-upload-queue for the upload flow documentation.

Right now this PR doesn't implement all of that (such as the approval part for example), as I am splitting things up into multiple PRs.

Once that's done, I will port over the documentation as well.

This is just the first PR for this experimental package, and since it's experimental, don't expect the documentation to be finished.

In the meantime, I highly recommend checking out the full code over at https://github.com/swissspidy/media-experiments/tree/main/packages/upload-media so you get a better understanding.

Reply:
It's ok for the documentation.

That said, If we're adding "selectors" about approvals and pending state but no actions to act on these, how does this work? Should we remove the features related to approvals... for now (actions and selectors added at the same time).

Reply:
Ah, I see that I removed the actions but kept the reducers & selectors. I'll add them back to make it less confusing.

But the actual usage of those actions will be added in separate PRs, as that involves UI changes in other packages, so way out of scope for this PR.